src/main/java/com/suse/salt/netapi/client/impl/HttpAsyncClientImpl.java

```diff
@@ -18,59 +16,47 @@
 import org.apache.http.nio.client.HttpAsyncClient;
 
 import java.net.URI;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 
 /**
- * Representation of a connection to Salt for issuing API requests using Apache's
- * HttpAsyncClient.
+ * AsyncHttpClient implemented with Apache's HttpAsyncClient.
  *
- * @param <T> type of result retrieved using this HTTP connection
  */
-public class HttpAsyncClientConnection<T> implements AsyncConnection<T> {
+public class HttpAsyncClientImpl implements AsyncHttpClient {
 
     /** HTTP client instance */
     private final HttpAsyncClient httpClient;
 
-    /** Endpoint */
-    private final String endpoint;
-
-    /** Configuration */
-    private final ClientConfig config;
-
-    /** Parser to parse the returned result */
-    private final JsonParser<T> parser;
+    public static HttpAsyncClient defaultClient() {
+        //TODO
+        return null;
```

Comment:
@lucidd: The default client can now be generated via `HttpClientUtils` as far as I understand, so should we remove the TODO and this whole method here?